File: leetcode/0205-isomorphic-strings/solution.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        vector<char> values;
        if (s.size()!=t.size()){
            return false;
        }
        unordered_map<char,char> mapping;
        for (int i=0;i<s.size();i++){
            auto it=mapping.find(s[i]);
            if(it != mapping.end()){
                if(it->second!=t[i]){
                    return false;
                }
            }
            else{
                mapping.insert({s[i],t[i]});
                if(find(values.begin(),values.end(),t[i])!=values.end()){
                    return false;
                }
                values.push_back(t[i]);
                
            }
        }
        return true;
    }
};

File: leetcode/0205-isomorphic-strings/solution.py
class Solution(object):
    def isIsomorphic(self, s, t):
        if (len(s)!=len(t)):
            return False
        map={}
        values=[]
        for i in range(len(s)):
            if (s[i] in map.keys()):
                if(map[s[i]]!=t[i]):
                    return False
            else:
                map[s[i]]=t[i]
                if (t[i] in values):
                    return False
                values.append(t[i])
        return True
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        
